### PR TITLE
STREAM-30 Template need both config and configDiff

### PR DESCRIPTION
**Pull request by akuma432**
* updated script command. made --config mandatory.
* Deleted Makefile-evolve-schema.schema.ssp. Updated existing Makefile.schema.ssp
* added copy-into.sql.ssp for schema evolution

### CODE REVIEW

File: src/main/java/io/phdata/streamliner/schemadefiner/util/TemplateFunction.java [hunk 1]
```diff
@@ -0,0 +1,150 @@
+package io.phdata.streamliner.schemadefiner.util;
```
Comment (akuma432) [inline src/main/java/io/phdata/streamliner/schemadefiner/util/TemplateFunction.java:1]: TemplateFunction class is used in MDT template. Hence added this class and few methods that are used in template.

File: src/main/resources/templates/snowflake/Makefile.schema.ssp [hunk 2]
```diff
@@ -14,11 +14,24 @@ See the License for the specific language governing permissions and
...
+    %>
+run-all: ${runTables}
+
+#for (table <- tableDiffs)
+run-${table.destinationName}:
+	$(MAKE) run -C ${table.destinationName}
+
+#end
+#elseif (tables != null)
+    <%
```
Comment (brockn) [inline src/main/resources/templates/snowflake/Makefile.schema.ssp:30]: I think on most runs, both tableDiffs AND tables are going to have data. Should this be in a seperate if, not an else-if?

File: src/main/resources/templates/snowflake/Makefile.schema.ssp [hunk 3]
```diff
@@ -14,11 +14,24 @@ See the License for the specific language governing permissions and
...
 --%>
 <%@ val tables: Seq[io.phdata.streamliner.schemadefiner.model.TableDefinition] %>
-<%
-	val firstRunTables = tables.map(t => s"first-run-${t.destinationName}").mkString(" ")
-	val runTables = tables.map(t => s"run-${t.destinationName}").mkString(" ")
-	val cleanTables = tables.map(t => s"clean-${t.destinationName}").mkString(" ")
-%>
+<%@ val tableDiffs: Seq[io.phdata.streamliner.schemadefiner.model.TableDiff] %>
+#if (!tableDiffs.isEmpty)
+    <%
```
Comment (brockn) [inline src/main/resources/templates/snowflake/Makefile.schema.ssp:19]: Does tableDiffs take care of the case where a new table is created inside the source database? If not, how will that case be handled?

File: src/main/resources/templates/snowflake/snowflake-schema-evolution/copy-into.sql.ssp [hunk 4]
```diff
@@ -0,0 +1,31 @@
+<%--
+Copyright 2018 phData Inc.
+
```
Comment (brockn) [inline src/main/resources/templates/snowflake/snowflake-schema-evolution/copy-into.sql.ssp:3]: There is a shared template for copy-into. Why aren't we using that here?

**Review by brockn — CHANGES_REQUESTED** [review]
A few questions below.

File: src/main/resources/templates/snowflake/Makefile.schema.ssp (hunk 2 above)
Comment (akuma432) [reply src/main/resources/templates/snowflake/Makefile.schema.ssp:30]: This is what I am thinking 
1. initial script to create table  -> conifg present configDiff absent
2. evolve schema to add new columns -> config present configDiff present 

For 2nd case, currently only evolve schema scripts are generated. Initial scripts are already generated during first run of script command. If we add separate if here then again whole initial scripts  and evolve schema scripts will be generated.

File: src/main/resources/templates/snowflake/Makefile.schema.ssp (hunk 3 above)
Comment (akuma432) [reply src/main/resources/templates/snowflake/Makefile.schema.ssp:19]: I believe creating a new table in source db is handled. Let me test it once. I will update here

File: src/main/resources/templates/snowflake/snowflake-schema-evolution/copy-into.sql.ssp (hunk 4 above)
Comment (akuma432) [reply src/main/resources/templates/snowflake/snowflake-schema-evolution/copy-into.sql.ssp:3]: My mistake. We can use the shared template

File: src/main/resources/templates/snowflake/Makefile.schema.ssp (hunk 3 above)
Comment (akuma432) [reply src/main/resources/templates/snowflake/Makefile.schema.ssp:19]: This case is handled.  I created a new table STREAMLINER_TEST2 in oracle and then performed schema evolution step.
Screenshot 2021-07-13 at 7 34 48 PM

File: src/main/resources/templates/snowflake/snowflake-schema-evolution/copy-into.sql.ssp (hunk 4 above)
Comment (akuma432) [reply src/main/resources/templates/snowflake/snowflake-schema-evolution/copy-into.sql.ssp:3]: I found one difference in the COPY INTO  query.
In shared template this is the query used : 
COPY INTO ${table.destinationName}_stg (${unescape(table.columnList(null))})
FROM ( SELECT ${unescape(sourceColumnStr)}
FROM @${configuration.name}_stage/${table.sourceName})
FILE_FORMAT = ( TYPE = PARQUET);

And I copied this file from snowflake-snowpipe-append folder. There this query is used :  
COPY INTO ${table.destinationName} (${unescape(table.columnList(null))})
FROM ( SELECT ${unescape(sourceColumnStr)}
FROM @${configuration.name}_stage/${table.sourceName})
FILE_FORMAT = ( TYPE = PARQUET);

There is difference in the table name. "_stg" is appended in the shared template. So far in all the tests I am able to load data with query present in this file.  I think if I use the shared template the query will fail.

**Review by brockn — APPROVED** [review]

**Pull request merged**